kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -132,9 +132,13 @@ private void updateStubs() {
     List<ListenableFuture<SearchResult<LogMessage>>> queryServers = new ArrayList<>(stubs.size());
     span.tag("queryServerCount", String.valueOf(stubs.size()));
 
+    long GRPC_QUERY_TIMEOUT_DURATION = READ_TIMEOUT_MS - 100;
```

Comment:
Hmm! i am not sure how we want this to be? Since you rightly brought up the fact that the test can override the static, we want `GRPC_QUERY_TIMEOUT_DURATION` to be evaluated  on every request? Is there any benefiting than just hardcoding it with `READ_TIMEOUT_MS - 100` in `withDeadlineAfter`?